dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -1112,6 +1112,150 @@ def median_abs_deviation(self):
+        bin_edges_pos = np.append([0], bin_edges[id_zero:])
+
+        bin_counts_neg = np.append([normalized_bin_counts[id_zero - 1] -
+             normalized_bin_counts[id_zero - 1] * (bin_edges[id_zero] /
+                (bin_edges[id_zero] - bin_edges[id_zero - 1]))],
+            normalized_bin_counts[:id_zero - 1][::-1])
+        bin_edges_neg = np.append([0], -bin_edges[:id_zero][::-1])
+
+        bin_counts_long, bin_edges_long = bin_counts_pos, bin_edges_pos
+        bin_counts_short, bin_edges_short = bin_counts_neg, bin_edges_neg
+        if len(bin_counts_pos) < len(bin_counts_neg):
+            bin_counts_long, bin_edges_long = bin_counts_neg, bin_edges_neg
+            bin_counts_short, bin_edges_short = bin_counts_neg, bin_edges_neg
+
+        for i in range(min(len(bin_counts_long), 1000)):
+            print(i, (bin_edges_long[i], bin_edges_long[i+1]), bin_counts_long[i])
+        for i in range(min(len(bin_counts_short), 1000)):
+            print(i, (bin_edges_short[i], bin_edges_short[i + 1]), bin_counts_short[i])
+        # iterate through two histograms until getting the points
+        # with cumsum counts as 0.5
+        cumsum_count, cumsum_count_prev = 0, 0
+        id = 0
+        bin_width = bin_edges_long[-1] - bin_edges_long[-2]
+        while id <= len(bin_counts_short) - 1 and cumsum_count < 0.5:
+            cumsum_count_prev = cumsum_count
+            cumsum_count += bin_counts_long[id] + bin_counts_short[id]
+            print(id, (bin_edges_long[i], bin_edges_long[i+1]), bin_counts_long[id],
+                  (bin_edges_short[i], bin_edges_short[i + 1]), bin_counts_short[id])
+            id += 1
+
+        # if cumsum count passes 0.5 in the middle, interpolate the median value
+        id -= 1
+        if cumsum_count >= 0.5:
+            if bin_edges_long[id] == bin_edges_short[id]:
+                return np.interp(0.5, [cumsum_count_prev, cumsum_count],
+                                 [bin_edges_long[id], bin_edges_long[id + 1]])
+            else:
+                edges_ahead = [bin_edges_long[id], bin_edges_long[id + 1]]
+                counts_ahead = bin_counts_long[id]
+                edges_behind = [bin_edges_short[id], bin_edges_short[id + 1]]
+                counts_behind = bin_counts_short[id]
+                if bin_edges_long[id] < bin_edges_short[id]:
+                    edges_ahead = [bin_edges_short[id], bin_edges_short[id + 1]]
+                    counts_ahead = bin_counts_short[id]
+                    edges_behind = [bin_edges_long[id], bin_edges_long[id + 1]]
+                    counts_behind = bin_counts_long[id]
+
+                # check if the left part includes the median
+                cumsum_count_left = cumsum_count_prev
+                if id > 1:
+                    cumsum_count_left = cumsum_count_prev + \
+                        (edges_ahead[0] - edges_behind[0]) / bin_width * \
+                        counts_behind
```

Comment:
I'm concerned this isn't correct, for example:
if `id==2` bc when short/long 2 were added, cumsum was > 0.5, the left part would actually be an overlap of short 2 and long 1. This would need a subtraction of part of long 1 and included in the interpolation. the interpolation would have to start from the trailing edge of short 2 (which it does) and go to the leading edge of long 1 (i'm not sure it is doing this). Additionally, cumsum_prev would have to have the subtraction as mentioned before and hte left would need the addition  relative to both.

Currently, it is including all of the long part but if it falls within it, then the long part needs to be interpolated too.

Reply:
somewhat like you said, but to be precise:
- I specify behind vs ahead bins and counts: behind hist is the one with lower edges for any index
- the left part is the part starting from cumsum_prev up to the edges_ahead[0]
- the counts of the left part is a proportion of counts_behind with the ratio given by that formular
I'm confident that is the correct one